framework/db/ActiveQuery.php

```diff
@@ -794,4 +805,64 @@ public function alias($alias)
         }
         return $this;
     }
+
+    /**
+     * Replace occurence of self::ALIAS_PLACEHOLDER by $aliasName into given conditions
+     * @param string|array $condition the given conditions
+     * @param string $aliasName
+     */
+    public function replaceAliasPlaceholder($condition, $aliasName)
+    {
+        if ((empty($condition)) or (is_int($condition))) {
+            return $condition;
+        } elseif (is_string($condition)) {
+            return str_replace(self::ALIAS_PLACEHOLDER, $aliasName, $condition);
+        } elseif (is_array($condition)) {
+            $conditionNew = [];
+            foreach ($condition as $key => $value) {
+                unset($condition[$key]);
+                $key = self::replaceAliasPlaceholder($key, $aliasName);
+                $value = self::replaceAliasPlaceholder($value, $aliasName);
+                $conditionNew[$key] = $value;
+            }
+            $condition = $conditionNew;
+        } elseif ($condition instanceof \yii\db\Expression) {
+            $condition->expression = self::replaceAliasPlaceholder($condition->expression, $aliasName);
+        }
+        return $condition;
+    }
+
+    /**
+     * Returns the table alias (or the table name if empty) of the current ActiveQuery
+     * @return string table alias or table name of the current activeQuery
+     */
+    public function getAliasTableName()
+    {
+        list (, $alias) = self::getQueryTableName($this);
```

Comment:
retun self::getQueryTableName($this)[1];

?

Reply:
Ok I changed it this way, but it's not the way it was done in the master.